In [2]:
%pip install Cython
%pip install PyChest
%pip install matplotlib
import PyChest
import numpy as np
import matplotlib.pyplot as plt
import gzip

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [2]:
text = "The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions. Artificial intelligence (AI) continues to revolutionize various industries, from healthcare to finance and beyond. AI algorithms, powered by machine learning and neural networks, analyze vast datasets to uncover patterns and make predictions. In healthcare, AI aids in diagnostics and personalized medicine, while in finance, it optimizes trading strategies and risk management. Ethical considerations surrounding AI's impact on employment and privacy are ongoing topics of debate. As AI technology advances, its integration into everyday life raises questions about regulation, transparency, and the societal implications of machine intelligence.	"

text1 = "The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions."

s_out = gzip.compress(text.encode())
print( "s_out : " , s_out)

s_out_octet = list(s_out)
text1_octet = list(gzip.compress(text1.encode()))

l = []
for i in range(len(text1_octet)):
    if text1_octet[i] == s_out_octet[i] : 
        l.append(0)
    else : 
        l.append((text1_octet[i],s_out_octet[i]))


print("s_out_octet : ", s_out_octet)
print( "text1_octet : ",text1_octet)
print("l : ", l ) 

s_out :  b"\x1f\x8b\x08\x00\xc6\xf1\xcdf\x02\xffmT\xcb\x8e\xe46\x0c<\xe7/x\xcb.\xd0\xe9\x0f\xc8m\x10\xecan9l>\x80\x96\xd861\xb2\xe8P\x92g\x9d\xafO\xd1\xdd\xf3\xc8\x03h\xa0\x01\x9b,\x16\xab\x8a\xfe\xbe\x08mnIZ#\xbb\xd1\xb6X\xb7v\xd4\xbeH\xd3F\xf8%\x1fI\xb9P7*z\x13\xb2J\xdf\xd8\xfbr\xa5\xdf\x0b\xd7\xde.\xc4ef\xc1_\xcd\x94\xc4;k\xa5\x89S\x17W\xa6\xd1\x84\xfa\x02\x98I--\xb2j\x02\xd4\xdb<@&\xab;z\x00=/\x9d\xa4\x8a\xcf\x07i\x8d7o\xd5\x8f\x87\xad\x9bK\xc6;\x9a\xcbH\x06\xdc\xd5\x8a\xa4Q\xa4]\xe9\xfb\xe26\xe6\xe5_\xf4/\x94\xd8'\x10\xcej?4K\xac\xc3S3\x9f\x00\x14t\xed\xc71K\x8d\xc7.E\xb8I\xbeP\x1b-V\xd0:\x83\xb8\xbc\x95\x14\xd9\xa54\xaa\x12\xc4\xd9\x0f\xba\x99\x13\x8b\xdb\xa4\t\xcdmS\xe7\xae\x18\x05~\\u\xe5r\xb2\n\tkw\xec\xd1?d\xd6\xba[\xd9\x05\xfb\xbf\x1a\xad!\x17&\xce\xd2~=\x07\x9eJ\xfc\x92e\x93\x9a\xa5v\x80CK \xc3\x9e\x94\x86{\x10CK\x94\xf6\xe5(\xfcb\x9ai\x95ur\xaerz\x98\x96bn[\xe1v\xba\x83=?p\xb5\xfe\x1f\xf2\x97\xdf\xb8\xec\x00MG*\xf2\xf5\r\xbeu\xb7\x95\xaf\xf4\x07\xca\x1d\x14k\x8e\xd9\xffM\x

In [13]:
utile = s_out_octet [10: -15 ]
print( "len(utile) : ", len(utile))
print(utile)


min_distance = 0.5
process_count = 2

print("avec le text initial en enlevant les octets inutiles de début et de fin : ", PyChest.find_changepoints(utile, min_distance, process_count))

print("avec le text initial : ", PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count))


# print(estimates)

len(utile) :  709
[109, 84, 203, 142, 228, 54, 12, 60, 231, 47, 120, 203, 46, 208, 233, 15, 200, 109, 16, 236, 97, 110, 57, 108, 62, 128, 150, 216, 54, 49, 178, 232, 80, 146, 103, 157, 175, 79, 209, 221, 243, 200, 3, 104, 160, 1, 155, 44, 22, 171, 138, 254, 190, 8, 109, 110, 73, 90, 35, 187, 209, 182, 88, 183, 118, 212, 190, 72, 211, 70, 248, 37, 31, 73, 185, 80, 55, 42, 122, 19, 178, 74, 223, 216, 251, 114, 165, 223, 11, 215, 222, 46, 196, 101, 102, 193, 95, 205, 148, 196, 59, 107, 165, 137, 83, 23, 87, 166, 209, 132, 250, 2, 152, 73, 45, 45, 178, 106, 2, 212, 219, 60, 64, 38, 171, 59, 122, 0, 61, 47, 157, 164, 138, 207, 7, 105, 141, 55, 111, 213, 143, 135, 173, 155, 75, 198, 59, 154, 203, 72, 6, 220, 213, 138, 164, 81, 164, 93, 233, 251, 226, 54, 230, 229, 95, 244, 47, 148, 216, 39, 16, 206, 106, 63, 52, 75, 172, 195, 83, 51, 159, 0, 20, 116, 237, 199, 49, 75, 141, 199, 46, 69, 184, 73, 190, 80, 27, 45, 86, 208, 58, 131, 184, 188, 149, 20, 217, 165, 52, 170, 18, 196, 217, 15, 186, 15

Computing Segment Scores:  100% |█████████████████████| (6/6, 854 it/s)
Finding Exact Changepoints:  100% |██████████████| (119/119, 3107 it/s)
Computing Segment Scores:  100% |█████████████████████| (6/6, 648 it/s)
Finding Exact Changepoints:  100% |██████████████| (123/123, 3299 it/s)


### Interprétation des Octets
Octets 0-1 : Signature du fichier GZIP, toujours 1f 8b en hexadécimal, qui correspond à [31, 139] en décimal. Cela identifie le fichier comme un fichier GZIP.

Octet 2 : Version de la compression, toujours 08 (qui correspond à [8]), indiquant que le format de compression est "deflate".

Octet 3 : Flags indiquant les options utilisées (souvent 00 si aucune option n'est activée).

Octets 4-7 : Horodatage de la création du fichier, représenté en format UNIX (le nombre de secondes depuis le 1er janvier 1970). Dans votre exemple, cela correspond aux valeurs [216, 165, 205, 102].

Octet 8 : Drapeau supplémentaire (souvent 02, ce qui signifie que la compression "deflate" est utilisée).

Octet 9 : Indique le système de fichiers d'origine, par exemple, [255] peut indiquer un système de fichiers non spécifié.

Octets 10 et suivants : Données compressées.

[31, 139] : Signature du fichier GZIP (0x1F 0x8B).

[8] : Compression avec "deflate".

[0] : Aucun flag activé (pas de nom de fichier original, pas de commentaire, etc.).

[216, 165, 205, 102] : Timestamp en hexadécimal (0x66cda5d8), qui est l'heure à laquelle la compression a été effectuée.

[2] : Indique que la compression "deflate" est utilisée.

[255] : Indique que le système de fichiers d'origine n'est pas spécifié.

[243, 201, 47, ... , 45, 74, 5] : Contenu des données compressées. Ces octets représentent les données "deflate".

[126, 117, 32, 142, 20, 0, 0, 0] : Pied de page contenant le CRC32 et la taille des données non compressées.



## On remarque que le change point detecté change uniquement de 10 mots magré les octets de début et de fin enlevé donc on garde le text entier initial

## Test pour plusieurs min_distances

In [14]:
text = "The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions. Artificial intelligence (AI) continues to revolutionize various industries, from healthcare to finance and beyond. AI algorithms, powered by machine learning and neural networks, analyze vast datasets to uncover patterns and make predictions. In healthcare, AI aids in diagnostics and personalized medicine, while in finance, it optimizes trading strategies and risk management. Ethical considerations surrounding AI's impact on employment and privacy are ongoing topics of debate. As AI technology advances, its integration into everyday life raises questions about regulation, transparency, and the societal implications of machine intelligence.	"

taille_echantillon = 10



a = len(text) // taille_echantillon

# print(a)
# for i in range(a):
#     text_test = text[0:i*taille_echantillon]
#     print(text_test)
#     text_test_octet = list(gzip.compress(text_test.encode()))
#     print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))


# min_distance = 0.3
process_count = 2

for i in np.arange(0.1, 0.6, 0.05):
    print(f"min_distance {i} : ", PyChest.find_changepoints(gzip.compress(text.encode()), i, process_count))



# estimates = PyChest.find_changepoints(utile, i, process_count)
# print(estimates)

734


Computing Segment Scores:  100% |██████████████████| (54/54, 7538 it/s)
Finding Exact Changepoints:  100% |██████████████| (288/288, 2251 it/s)        


min_distance 0.1 :  [39]
min_distance 0.15000000000000002 :  [85]


Computing Segment Scores:  100% |██████████████████| (34/34, 3923 it/s)
Finding Exact Changepoints:  100% |██████████████| (259/259, 2602 it/s)        
Computing Segment Scores:  100% |██████████████████| (24/24, 2695 it/s)        
Finding Exact Changepoints:  100% |██████████████| (245/245, 2746 it/s)        
Computing Segment Scores:  100% |██████████████████| (18/18, 1719 it/s)        
Finding Exact Changepoints:  100% |██████████████| (186/186, 3131 it/s)


min_distance 0.20000000000000004 :  [66]
min_distance 0.25000000000000006 :  [213]


Computing Segment Scores:  100% |██████████████████| (14/14, 1579 it/s)
Finding Exact Changepoints:  100% |██████████████| (296/296, 3587 it/s)        
Computing Segment Scores:  100% |██████████████████| (12/12, 1371 it/s)        


min_distance 0.30000000000000004 :  [98]
min_distance 0.3500000000000001 :  [639]


Finding Exact Changepoints:  100% |██████████████| (258/258, 3246 it/s)        
Computing Segment Scores:  100% |█████████████████████| (9/9, 982 it/s)        
Finding Exact Changepoints:  100% |██████████████| (294/294, 3237 it/s)        


min_distance 0.40000000000000013 :  [137]
min_distance 0.45000000000000007 :  [387]
min_distance 0.5000000000000001 :  [422]
min_distance 0.5500000000000002 :  [434]


Computing Segment Scores:  100% |████████████████████| (8/8, 1024 it/s)
Finding Exact Changepoints:  100% |██████████████| (220/220, 3067 it/s)        
Computing Segment Scores:  100% |█████████████████████| (6/6, 682 it/s)        
Finding Exact Changepoints:  100% |██████████████| (123/123, 3634 it/s)
Computing Segment Scores:  100% |█████████████████████| (5/5, 371 it/s)
Finding Exact Changepoints:  100% |██████████████| (135/135, 2545 it/s)


In [ ]:
text = "The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions. Artificial intelligence (AI) continues to revolutionize various industries, from healthcare to finance and beyond. AI algorithms, powered by machine learning and neural networks, analyze vast datasets to uncover patterns and make predictions. In healthcare, AI aids in diagnostics and personalized medicine, while in finance, it optimizes trading strategies and risk management. Ethical considerations surrounding AI's impact on employment and privacy are ongoing topics of debate. As AI technology advances, its integration into everyday life raises questions about regulation, transparency, and the societal implications of machine intelligence.	"

text_octet = gzip.compress(tex.encode())




estimates = PyChest.find_changepoints(text_octet, , process_count)
print(estimates)

## Verification du change points avec celui estimé

In [3]:
text = "The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions. Artificial intelligence (AI) continues to revolutionize various industries, from healthcare to finance and beyond. AI algorithms, powered by machine learning and neural networks, analyze vast datasets to uncover patterns and make predictions. In healthcare, AI aids in diagnostics and personalized medicine, while in finance, it optimizes trading strategies and risk management. Ethical considerations surrounding AI's impact on employment and privacy are ongoing topics of debate. As AI technology advances, its integration into everyday life raises questions about regulation, transparency, and the societal implications of machine intelligence.	"

taille_echantillon = 10

a = len(text) // taille_echantillon

print(a)
for i in range(a):
    text_test = text[0:i*taille_echantillon]
    print(text_test)
    text_test_octet = list(gzip.compress(text_test.encode()))
    print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))


min_distance = 0.5
process_count = 2

# for i in np.arange(0.1, 1, 0.05):
#     print(f"min_distance {i} : ", PyChest.find_changepoints(utile, i, process_count))



estimates = PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count)
print(estimates)

134

20
The proces
30
The process of photo
40
The process of photosynthesis 
50
The process of photosynthesis is crucial
58
The process of photosynthesis is crucial to life o
68
The process of photosynthesis is crucial to life on Earth. P
78
The process of photosynthesis is crucial to life on Earth. Plants, alg
84
The process of photosynthesis is crucial to life on Earth. Plants, algae, and ce
90
The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bact
95
The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use t
100
The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this bioche
108
The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical proc
114
The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to con
118
The process of photosynthe

Computing Segment Scores:  100% |█████████████████████| (6/6, 503 it/s)
Finding Exact Changepoints:  100% |██████████████| (123/123, 1561 it/s)        


## Les résultats sont plutot satisfaisants, le change points estimé (422) correspond à peu près à la position, dans le text compressé, du change points. 

# Nouveau test avec le meme texte mais rallongé plus grand

In [42]:
text =" The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process involves two main stages: the light-dependent reactions occurring in the thylakoid membranes of chloroplasts, and the light-independent reactions (Calvin cycle) in the stroma. Understanding photosynthesis is fundamental not only to biology but also to our understanding of climate regulation and sustainable energy solutions. Photosynthesis research continues to evolve, revealing more about how plants adapt to changing environments and how we might harness this knowledge to improve crop yields and develop biofuels. Moreover, the study of photosynthetic mechanisms offers insights into addressing global challenges such as food security and renewable energy. Artificial intelligence (AI) continues to revolutionize various industries, from healthcare to finance and beyond. AI algorithms, powered by machine learning and neural networks, analyze vast datasets to uncover patterns and make predictions. In healthcare, AI aids in diagnostics and personalized medicine, enhancing the accuracy and efficiency of treatments. In finance, AI optimizes trading strategies, improves risk management, and even helps in detecting fraudulent activities. The rapid advancement of AI technology raises important ethical questions about its impact on employment, privacy, and decision-making. As AI becomes increasingly integrated into everyday life, society must grapple with issues of regulation, transparency, and the long-term implications of machine intelligence. The potential for AI to reshape industries and human experiences is vast, but it also requires careful consideration of how these technologies are developed and deployed."

min_distance = 0.4
process_count = 2

taille_echantillon = 100

estimates = PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count)
print(estimates)

a = len(text) // taille_echantillon

print(a)
for i in range(a):
    text_test = text[0:i*taille_echantillon]
    print(text_test)
    text_test_octet = list(gzip.compress(text_test.encode()))
    print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))




# for i in np.arange(0.1, 1, 0.05):
#     print(f"min_distance {i} : ", PyChest.find_changepoints(utile, i, process_count))



estimates = PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count)
print(estimates)

Computing Segment Scores:  100% |█████████████████████| (9/9, 623 it/s)
Finding Exact Changepoints:  100% |██████████████| (278/278, 2737 it/s)        
Computing Segment Scores:  100% |█████████████████████| (9/9, 384 it/s)        


[626]
19

20
 The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use 
100
 The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. T
154
 The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen leve
205
 The process of photosynthesis is crucial to life on Earth. Plants, algae, and certain bacteria use this biochemical process to convert light energy into chemical energy stored in glucose molecules. Through photosynthesis, carbon dioxide is absorbed and oxygen is released, sustaining the oxygen levels necessary for aerobic respiration in animals. This intricate process 

Finding Exact Changepoints:  100% |██████████████| (278/278, 2639 it/s)        


In [15]:
taille_echantillon = 10



a = len(text) // taille_echantillon

# print(a)
# for i in range(a):
#     text_test = text[0:i*taille_echantillon]
#     print(text_test)
#     text_test_octet = list(gzip.compress(text_test.encode()))
#     print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))


# min_distance = 0.3
process_count = 2

for i in np.arange(0.1, 0.6, 0.05):
    print(f"min_distance {i} : ", PyChest.find_changepoints(gzip.compress(text.encode()), i, process_count))



# estimates = PyChest.find_changepoints(utile, i, process_count)
# print(estimates)

1044


Computing Segment Scores:  100% |██████████████████| (54/54, 5318 it/s)
Finding Exact Changepoints:  100% |██████████████| (442/442, 2183 it/s)        
Computing Segment Scores:  100% |██████████████████| (34/34, 2643 it/s)        


min_distance 0.1 :  [400, 603, 852, 959]


Finding Exact Changepoints:  100% |██████████████| (371/371, 2740 it/s)        
Computing Segment Scores:  100% |██████████████████| (24/24, 2015 it/s)        


min_distance 0.15000000000000002 :  [70]


Finding Exact Changepoints:  100% |██████████████| (350/350, 2848 it/s)        
Computing Segment Scores:  100% |██████████████████| (18/18, 1585 it/s)        


min_distance 0.20000000000000004 :  [510]
min_distance 0.25000000000000006 :  [912]


Finding Exact Changepoints:  100% |██████████████| (348/348, 3188 it/s)        
Computing Segment Scores:  100% |██████████████████| (14/14, 1038 it/s)        
Finding Exact Changepoints:  100% |██████████████| (416/416, 3039 it/s)        
Computing Segment Scores:  100% |██████████████████| (12/12, 1130 it/s)        
Finding Exact Changepoints:  100% |██████████████| (366/366, 3235 it/s)        
Computing Segment Scores:  100% |█████████████████████| (9/9, 878 it/s)        


min_distance 0.30000000000000004 :  [239]
min_distance 0.3500000000000001 :  [654]


Finding Exact Changepoints:  100% |██████████████| (278/278, 2865 it/s)        
Computing Segment Scores:  100% |█████████████████████| (8/8, 715 it/s)        
Finding Exact Changepoints:  100% |██████████████| (314/314, 3325 it/s)        
Computing Segment Scores:  100% |█████████████████████| (6/6, 713 it/s)        


min_distance 0.40000000000000013 :  [626]
min_distance 0.45000000000000007 :  [521]
min_distance 0.5000000000000001 :  [768]
min_distance 0.5500000000000002 :  [437]


Finding Exact Changepoints:  100% |██████████████| (348/348, 2881 it/s)        
Computing Segment Scores:  100% |█████████████████████| (5/5, 559 it/s)        
Finding Exact Changepoints:  100% |██████████████| (382/382, 2974 it/s)        


## Nouveau text sur d'autres sujets 

In [44]:
text = "The history of ancient Egypt is a fascinating journey through one of the most advanced civilizations of the ancient world. From the construction of the colossal pyramids to the complex religious beliefs centered around the gods and the afterlife, Egypt's contributions to art, architecture, and governance were monumental. The pharaohs, regarded as both divine rulers and political leaders, played a crucial role in maintaining the kingdom's stability and prosperity. Hieroglyphics, the ancient writing system, not only recorded historical events but also provided insights into the daily lives, culture, and spirituality of the Egyptians. The Nile River was the lifeblood of the civilization, providing water, fertile land, and a means of transport that enabled trade and communication. The discovery of the Rosetta Stone in 1799 was pivotal in deciphering Egyptian scripts, opening a window into understanding this ancient civilization more profoundly. The phenomenon of black holes continues to intrigue scientists and astronomers around the world. These mysterious regions in space, where gravity is so intense that nothing, not even light, can escape, challenge our understanding of physics. Black holes are formed when massive stars collapse under their own gravity, creating a singularity—a point of infinite density. The event horizon marks the boundary beyond which nothing can return, making black holes invisible to direct observation. However, their presence can be inferred through the effects on nearby stars and the bending of light, known as gravitational lensing. Recent advancements in technology, such as the Event Horizon Telescope, have allowed scientists to capture the first-ever image of a black hole, further deepening our understanding of these cosmic phenomena. Despite these advances, many questions remain about the nature of black holes, their role in the universe, and what happens inside them—questions that continue to push the boundaries of human knowledge."

In [45]:
taille_echantillon = 100


min_distance = 0.35
process_count = 2

estimates = PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count)
print(estimates)

a = len(text) // taille_echantillon

print(a)
for i in range(a):
    text_test = text[0:i*taille_echantillon]
    print(text_test)
    text_test_octet = list(gzip.compress(text_test.encode()))
    print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))




# for i in np.arange(0.1, 1, 0.05):
#     print(f"min_distance {i} : ", PyChest.find_changepoints(utile, i, process_count))



# estimates = PyChest.find_changepoints(gzip.compress(text.encode()), min_distance, process_count)
# print(estimates)

[514]
19

20
The history of ancient Egypt is a fascinating journey through one of the most advanced civilizations
99
The history of ancient Egypt is a fascinating journey through one of the most advanced civilizations of the ancient world. From the construction of the colossal pyramids to the complex religious belie
151
The history of ancient Egypt is a fascinating journey through one of the most advanced civilizations of the ancient world. From the construction of the colossal pyramids to the complex religious beliefs centered around the gods and the afterlife, Egypt's contributions to art, architecture, and gover
205
The history of ancient Egypt is a fascinating journey through one of the most advanced civilizations of the ancient world. From the construction of the colossal pyramids to the complex religious beliefs centered around the gods and the afterlife, Egypt's contributions to art, architecture, and governance were monumental. The pharaohs, regarded as both divine rulers and p

Computing Segment Scores:  100% |███████████████████| (12/12, 635 it/s)
Finding Exact Changepoints:  100% |██████████████| (238/238, 3024 it/s)        


In [28]:
taille_echantillon = 10



a = len(text) // taille_echantillon

# print(a)
# for i in range(a):
#     text_test = text[0:i*taille_echantillon]
#     print(text_test)
#     text_test_octet = list(gzip.compress(text_test.encode()))
#     print(len(text_test_octet))
    
print(len(list(gzip.compress(text.encode()))))


# min_distance = 0.3
process_count = 2

for i in np.arange(0.1, 0.6, 0.05):
    print(f"min_distance {i} : ", PyChest.find_changepoints(gzip.compress(text.encode()), i, process_count))



# estimates = PyChest.find_changepoints(utile, i, process_count)
# print(estimates)

1024


Computing Segment Scores:  100% |██████████████████| (54/54, 3411 it/s)
Finding Exact Changepoints:  100% |██████████████| (374/374, 2095 it/s)        


min_distance 0.1 :  [86, 943]


Computing Segment Scores:  100% |██████████████████| (34/34, 2191 it/s)
Finding Exact Changepoints:  100% |██████████████| (408/408, 2236 it/s)        


min_distance 0.15000000000000002 :  [76]
min_distance 0.20000000000000004 :  [107]


Computing Segment Scores:  100% |██████████████████| (24/24, 2335 it/s)
Finding Exact Changepoints:  100% |██████████████| (340/340, 3165 it/s)        
Computing Segment Scores:  100% |██████████████████| (18/18, 1560 it/s)        
Finding Exact Changepoints:  100% |██████████████| (425/425, 3429 it/s)        
Computing Segment Scores:  100% |██████████████████| (14/14, 1363 it/s)        
Finding Exact Changepoints:   54% |███████       | (168/309, 2569 it/s) [0s:0s]

min_distance 0.25000000000000006 :  [316]
min_distance 0.30000000000000004 :  [752]


Finding Exact Changepoints:  100% |██████████████| (309/309, 3023 it/s)        
Computing Segment Scores:  100% |███████████████████| (12/12, 713 it/s)        
Finding Exact Changepoints:  100% |██████████████| (238/238, 2610 it/s)        
Computing Segment Scores:  100% |█████████████████████| (9/9, 646 it/s)        


min_distance 0.3500000000000001 :  [514]
min_distance 0.40000000000000013 :  [316]


Finding Exact Changepoints:  100% |██████████████| (272/272, 2823 it/s)        
Computing Segment Scores:  100% |█████████████████████| (8/8, 586 it/s)        
Finding Exact Changepoints:  100% |██████████████| (308/308, 2671 it/s)        


min_distance 0.45000000000000007 :  [813]
min_distance 0.5000000000000001 :  [229]


Computing Segment Scores:  100% |█████████████████████| (6/6, 573 it/s)
Finding Exact Changepoints:  100% |██████████████| (342/342, 3007 it/s)        
Computing Segment Scores:  100% |█████████████████████| (5/5, 188 it/s)        


min_distance 0.5500000000000002 :  [752]


Finding Exact Changepoints:  100% |██████████████| (188/188, 2722 it/s)        
